packages/nimble-components/src/tooltip/specs/README.md

```diff
@@ -14,6 +14,102 @@ The nimble-tooltip project will first be implemented as a prototype, open issues
+- [Atlassian Design System](https://atlassian.design/components/tooltip/examples) - Tooltip component takes anchor element as content
+- [Adobe Spectrum (React)](https://react-spectrum.adobe.com/react-spectrum/Tooltip.html) - Separate wrapper component around tooltip and anchor elements.
+- [FAST](https://explore.fast.design/components/fast-tooltip) - Tooltip associated with anchor element by id.
+
+### Associate by `id`
+This is the FAST approach, and the current implementation. The tooltip has an `anchor` attribute which takes the `id` of the anchor element.
+```html
+<nimble-tooltip anchor="previewButton">...</nimble-tooltip>
+<nimble-button id="previewButton">Preview</nimble-button>
+```
+
+**Pros**
+- No additional implementation cost
+- Familiar pattern, e.g. `<label>` element with `for` attribute, `aria-labelledby` attribute, `aria-describedby` attribute
+- Reasonable DOM structure, i.e. tooltip sibling of anchor element
+
+**Cons**
+- Annoying to have to give anchor element an `id`
+- <span style="color:red">**Not conducive to use in shared components because `id` values must be unique**</span>
+
+### Tooltip slotted in anchor element
+The anchor element defines a `slot` for the tooltip and takes the tooltip as content.
+```html
+<nimble-button>
+    Preview
+    <nimble-tooltip slot="tooltip">...</nimble-tooltip>
+</nimble-button>
+```
+
+**Pros**
+- Familiar slotting pattern
+- Logical DOM structure (element contains its tooltip)
+- May not need `nimble-tooltip` element at all -- could slot tooltip content directly without wrapping in `nimble-tooltip` element
+
+**Cons**
+- Expensive to implement and maintain. <span style="color:red">**Requires forking and modifying the template of every component that wants to support a tooltip.**</span> (Though code duplication could be avoided by using shared template fragments.) Tooltip implementation leaks into every other component.
+- Cannot apply tooltip to native elements, like `<div>`
+
+### Tooltip implemented by each component
+There is no tooltip component, but instead, each component that wants to support a tooltip makes it a part of its template. Those components have an attribute like `tooltip` that takes the text to display in the tooltip. If the tooltip needs to support more than just unformated text, that approach would be a variation of "Tooltip slotted in anchor element", where there is no explicit tooltip element.
+```html
+<nimble-button tooltip="This is tooltip text">Preview</nimble-button>
+```
+
+**Pros**
+- Simple, intuitive API (similar to existing `title` attribute)
+- No dedicated tooltip component
+
+**Cons**
+- Expensive to implement and maintain. <span style="color:red">**Requires forking and modifying the template of every component that wants to support a tooltip.**</span> (Though code duplication could be avoided by using shared template fragments.) Cannot leverage FAST implementation?
```

Comment:
I'm not sure this is true. We can easily add new attributes to controls and make shared patterns for adding attributes. It's also very possible that we don't necessitate a template change in tooltip logic. The tooltip logic could be very standalone, ie register listeners on elements in the template and show the tooltip dynamically (document.createElement the tooltip and insert it).